docs/en/guides/pipeline-integrations/development/sgtk-developer-engine.md

```diff
@@ -0,0 +1,130 @@
+---
+layout: default
+title: Developing engines
+pagename: sgtk-developer-engine
+lang: en
+---
+
+# Developing your own engine
+
+## Introduction
+This document outlines some of the technical details relating to Toolkit engine development.
+
+Table of Contents:
+- [What is a Toolkit engine?](#what-is-a-toolkit-engine)
+- [Things to know before you start](#things-to-know-before-you-start)
+- [Approaches to engine integration](#approaches-to-engine-integration)
+    - [Host application includes Qt, PyQt/PySide and Python](#host-application-includes-qt-pyqtpyside-and-python)
+    - [Host application includes Qt and Python but not PySide/PyQt](#host-application-includes-qt-and-python-but-not-pysidepyqt)
+    - [Host application includes Python](#host-application-includes-python)
+    - [Host application does not contain Python but you can write plugins](#host-application-does-not-contain-python-but-you-can-write-plugins)
+    - [Host application provides no scriptability at all](#host-application-provides-no-scriptability-at-all)
+- [Qt window parenting](#qt-window-parenting)
+- [Startup behavior](#startup-behavior)
+- [Host application wish list](#host-application-wish-list)
+
+## What is a Toolkit engine?
+When developing an engine, you effectively establish a bridge between the host application and the various apps and frameworks that are loaded into the engine. 
+The engine makes it possible to abstract the differences between applications so that apps can be written in more of a software agnostic manner using Python and Qt.
+
+The engine is a collection of files, [similar in structure to an app](sgtk-developer-app.md#anatomy-of-the-template-starter-app). It has an `engine.py` file and this must derive from the core `Engine` [Base class](https://github.com/shotgunsoftware/tk-core/blob/master/python/tank/platform/engine.py). 
+Different engines then re-implement various aspects of this base class depending on their internal complexity. 
+An engine typically handles or provides the following services:
+
+- Menu management. At engine startup, once the apps have been loaded, the engine needs to create its Shotgun menu and add the various apps to this menu.
+- Logging methods are typically overridden to write to the application log/console.
+- Methods for displaying UI dialogs and windows. These methods are usually overridden, if the way the engine handles Qt is different from the default base class behavior, to ensure seamless integration of windows launched by Toolkit apps and the underlying host application window management setup.
+- Provides a `commands` dictionary containing all the command objects registered by apps. This is typically accessed when menu entries are created.
+- The base class exposes various init and destroy methods that are executed at various points in the startup process. These can be overridden to control startup and shutdown execution.
```

Comment:
There isn't actually an `init` method, but there are methods with `init` in the name.
There is a `destroy` method, but it think it is referring to the destroy methods as a collective rather than the one specifically called destroy.
So I would opt to not backtick it, but don't mind too much.